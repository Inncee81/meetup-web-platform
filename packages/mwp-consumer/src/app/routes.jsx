import React from 'react';

const Div = ({ children }) =>
	<div>
		{children}
	</div>;

const routes = [
	{
		path: '/',
		component: Div,
		query: [],
		indexRoute: {
			component: Div,
		},
		routes: [
			{
				path: 'foo',
				component: Div,
			},
		],
	},
];

export default routes;
