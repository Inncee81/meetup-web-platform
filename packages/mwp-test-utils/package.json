{
  "name": "mwp-test-utils",
  "version": "0.0.1",
  "description": "Helpers for running unit tests in MWP apps",
  "engineStrict": true,
  "engines": {
    "node": ">=8.1.0"
  },
  "main": "./lib/index.js",
  "scripts": {
    "build": "babel src --out-dir lib --ignore test.js,test.jsx --source-maps",
    "flow": "flow",
    "lint": "eslint . --ext .js,.jsx --fix",
    "test": "yarn run test:unit",
    "test:unit": "NODE_ENV=test jest --config jest.config.json",
    "test:watch": "yarn test:unit -- --watch"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/meetup/meetup-web-platform.git"
  },
  "author": "Meetup",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/meetup/meetup-web-platform/issues"
  },
  "homepage": "https://github.com/meetup/meetup-web-platform#readme",
  "dependencies": {
    "hapi": "16.5.2",
    "inert": "4.2.1",
    "meetup-web-mocks": "1.0.206",
    "mwp-cli": "1.1.129",
    "mwp-store": ">=0.0.1",
    "mwp-router": ">=0.0.1",
    "mwp-logger-plugin": ">=0.0.1",
    "react-helmet": "5.2.0",
    "redux": "3.7.2",
    "redux-observable": "0.16.0",
    "rxjs": "5.4.3",
    "tough-cookie": "2.3.2"
  },
  "devDependencies": {
    "flow-bin": "0.49.1",
    "jest": "20.0.4"
  }
}