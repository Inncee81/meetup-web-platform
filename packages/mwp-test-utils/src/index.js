import Hapi from 'hapi';
import Cookie from 'tough-cookie';

import { properties as serverConfig } from 'mwp-config/server';

export const MOCK_LOGGER = {
	debug: jest.fn(),
	info: jest.fn(),
	warn: jest.fn(),
	error: jest.fn(),
};

export const createFakeStore = fakeData => ({
	getState() {
		return fakeData;
	},
	dispatch() {},
	subscribe() {},
});

export const middlewareDispatcher = middleware => (storeData, action) => {
	let dispatched = null;
	const dispatch = middleware(createFakeStore(storeData))(
		actionAttempt => (dispatched = actionAttempt)
	);
	dispatch(action);
	return dispatched;
};

export const parseCookieHeader = cookieHeader => {
	const cookies =
		cookieHeader instanceof Array
			? cookieHeader.map(Cookie.parse)
			: [Cookie.parse(cookieHeader)];

	return cookies.reduce(
		(acc, cookie) => ({ ...acc, [cookie.key]: cookie.value }),
		{}
	);
};

export const getServer = () => {
	const config = { ...serverConfig, supportedLangs: ['en-US'] };

	const server = Hapi.server({
		port: 0,
		app: config,
	});

	server.app = {
		logger: MOCK_LOGGER,
	};

	server.plugins = {
		'mwp-api-proxy-plugin': {
			duotoneUrls: [],
		},
	};

	server.decorate('request', 'trackActivity', () => ({}));
	server.decorate('request', 'getLangPrefixPath', () => '/');
	server.decorate('request', 'getLanguage', () => 'en-US');

	server.logger = () => MOCK_LOGGER;

	server.ext('onPreHandler', (request, reply) => {
		request.plugins.tracking = {};
		reply.continue();
	});

	return server;
};

const IDENTITY_REDUCER = state => state;
export function testCreateStore(createStoreFn) {
	it('creates a store with store functions', () => {
		const basicStore = createStoreFn(IDENTITY_REDUCER);
		expect(basicStore.getState).toEqual(jasmine.any(Function));
		expect(basicStore.dispatch).toEqual(jasmine.any(Function));
	});
	it('creates a store with supplied initialState', done => {
		const initialState = { foo: 'bar' };
		const basicStore = createStoreFn(IDENTITY_REDUCER, initialState);
		basicStore.subscribe(() => {
			expect(basicStore.getState()).toEqual(initialState);
			done();
		});
		basicStore.dispatch({ type: 'dummy' });
	});
}
