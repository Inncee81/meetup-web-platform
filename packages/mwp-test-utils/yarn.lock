# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
# yarn lockfile v1


accept@^2.1.4:
  version "2.1.4"
  resolved "https://registry.yarnpkg.com/accept/-/accept-2.1.4.tgz#887af54ceee5c7f4430461971ec400c61d09acbb"
  dependencies:
    boom "5.x.x"
    hoek "4.x.x"

ammo@2.x.x, ammo@^2.0.4:
  version "2.0.4"
  resolved "https://registry.yarnpkg.com/ammo/-/ammo-2.0.4.tgz#bf80aab211698ea78f63ef5e7f113dd5d9e8917f"
  dependencies:
    boom "5.x.x"
    hoek "4.x.x"

asap@~2.0.3:
  version "2.0.6"
  resolved "https://registry.yarnpkg.com/asap/-/asap-2.0.6.tgz#e50347611d7e690943208bbdafebcbc2fb866d46"

b64@3.x.x:
  version "3.0.3"
  resolved "https://registry.yarnpkg.com/b64/-/b64-3.0.3.tgz#36afeee0d9345f046387ce6de8a6702afe5bb56e"

base64url@2.0.0, base64url@^2.0.0:
  version "2.0.0"
  resolved "https://registry.yarnpkg.com/base64url/-/base64url-2.0.0.tgz#eac16e03ea1438eff9423d69baa36262ed1f70bb"

bluebird@^3.4.6:
  version "3.5.1"
  resolved "https://registry.yarnpkg.com/bluebird/-/bluebird-3.5.1.tgz#d9551f9de98f1fcda1e683d17ee91a0602ee2eb9"

boom@5.x.x, boom@^5.2.0:
  version "5.2.0"
  resolved "https://registry.yarnpkg.com/boom/-/boom-5.2.0.tgz#5dd9da6ee3a5f302077436290cb717d3f4a54e02"
  dependencies:
    hoek "4.x.x"

boom@^4.0.0:
  version "4.3.1"
  resolved "https://registry.yarnpkg.com/boom/-/boom-4.3.1.tgz#4f8a3005cb4a7e3889f749030fd25b96e01d2e31"
  dependencies:
    hoek "4.x.x"

buffer-equal-constant-time@1.0.1:
  version "1.0.1"
  resolved "https://registry.yarnpkg.com/buffer-equal-constant-time/-/buffer-equal-constant-time-1.0.1.tgz#f8e71132f7ffe6e01a5c9697a4c6f3e48d5cc819"

call@^4.0.2:
  version "4.0.2"
  resolved "https://registry.yarnpkg.com/call/-/call-4.0.2.tgz#df76f5f51ee8dd48b856ac8400f7e69e6d7399c4"
  dependencies:
    boom "5.x.x"
    hoek "4.x.x"

catbox-memory@^2.0.4:
  version "2.0.4"
  resolved "https://registry.yarnpkg.com/catbox-memory/-/catbox-memory-2.0.4.tgz#433e255902caf54233d1286429c8f4df14e822d5"
  dependencies:
    hoek "4.x.x"

catbox@^7.1.5:
  version "7.1.5"
  resolved "https://registry.yarnpkg.com/catbox/-/catbox-7.1.5.tgz#c56f7e8e9555d27c0dc038a96ef73e57d186bb1f"
  dependencies:
    boom "5.x.x"
    hoek "4.x.x"
    joi "10.x.x"

content@3.x.x:
  version "3.0.6"
  resolved "https://registry.yarnpkg.com/content/-/content-3.0.6.tgz#9c2e301e9ae515ed65a4b877d78aa5659bb1b809"
  dependencies:
    boom "5.x.x"

core-js@^1.0.0:
  version "1.2.7"
  resolved "https://registry.yarnpkg.com/core-js/-/core-js-1.2.7.tgz#652294c14651db28fa93bd2d5ff2983a4f08c636"

cryptiles@3.x.x, cryptiles@^3.1.2:
  version "3.1.2"
  resolved "https://registry.yarnpkg.com/cryptiles/-/cryptiles-3.1.2.tgz#a89fbb220f5ce25ec56e8c4aa8a4fd7b5b0d29fe"
  dependencies:
    boom "5.x.x"

deep-equal@^1.0.1:
  version "1.0.1"
  resolved "https://registry.yarnpkg.com/deep-equal/-/deep-equal-1.0.1.tgz#f5d260292b660e084eff4cdbc9f08ad3247448b5"

ecdsa-sig-formatter@1.0.9:
  version "1.0.9"
  resolved "https://registry.yarnpkg.com/ecdsa-sig-formatter/-/ecdsa-sig-formatter-1.0.9.tgz#4bc926274ec3b5abb5016e7e1d60921ac262b2a1"
  dependencies:
    base64url "^2.0.0"
    safe-buffer "^5.0.1"

electrode-csrf-jwt@1.2.1:
  version "1.2.1"
  resolved "https://registry.yarnpkg.com/electrode-csrf-jwt/-/electrode-csrf-jwt-1.2.1.tgz#1006eb06bced930c82b7002fcd4be5b53cced807"
  dependencies:
    bluebird "^3.4.6"
    boom "^4.0.0"
    jsonwebtoken "^7.1.9"
    uuid "^2.0.1"

encoding@^0.1.11:
  version "0.1.12"
  resolved "https://registry.yarnpkg.com/encoding/-/encoding-0.1.12.tgz#538b66f3ee62cd1ab51ec323829d1f9480c74beb"
  dependencies:
    iconv-lite "~0.4.13"

exenv@^1.2.1:
  version "1.2.2"
  resolved "https://registry.yarnpkg.com/exenv/-/exenv-1.2.2.tgz#2ae78e85d9894158670b03d47bec1f03bd91bb9d"

fbjs@^0.8.16:
  version "0.8.16"
  resolved "https://registry.yarnpkg.com/fbjs/-/fbjs-0.8.16.tgz#5e67432f550dc41b572bf55847b8aca64e5337db"
  dependencies:
    core-js "^1.0.0"
    isomorphic-fetch "^2.1.1"
    loose-envify "^1.0.0"
    object-assign "^4.1.0"
    promise "^7.1.1"
    setimmediate "^1.0.5"
    ua-parser-js "^0.7.9"

hapi@16.5.2:
  version "16.5.2"
  resolved "https://registry.yarnpkg.com/hapi/-/hapi-16.5.2.tgz#d1dadf33721c6ac3aaa905ce086d9c7ffb883092"
  dependencies:
    accept "^2.1.4"
    ammo "^2.0.4"
    boom "^5.2.0"
    call "^4.0.2"
    catbox "^7.1.5"
    catbox-memory "^2.0.4"
    cryptiles "^3.1.2"
    heavy "^4.0.4"
    hoek "^4.2.0"
    iron "^4.0.5"
    items "^2.1.1"
    joi "^10.6.0"
    mimos "^3.0.3"
    podium "^1.3.0"
    shot "^3.4.2"
    statehood "^5.0.3"
    subtext "^5.0.0"
    topo "^2.0.2"

heavy@^4.0.4:
  version "4.0.4"
  resolved "https://registry.yarnpkg.com/heavy/-/heavy-4.0.4.tgz#36c91336c00ccfe852caa4d153086335cd2f00e9"
  dependencies:
    boom "5.x.x"
    hoek "4.x.x"
    joi "10.x.x"

hoek@2.x.x:
  version "2.16.3"
  resolved "https://registry.yarnpkg.com/hoek/-/hoek-2.16.3.tgz#20bb7403d3cea398e91dc4710a8ff1b8274a25ed"

hoek@4.x.x, hoek@^4.2.0:
  version "4.2.0"
  resolved "https://registry.yarnpkg.com/hoek/-/hoek-4.2.0.tgz#72d9d0754f7fe25ca2d01ad8f8f9a9449a89526d"

iconv-lite@~0.4.13:
  version "0.4.19"
  resolved "https://registry.yarnpkg.com/iconv-lite/-/iconv-lite-0.4.19.tgz#f7468f60135f5e5dad3399c0a81be9a1603a082b"

inert@4.2.1:
  version "4.2.1"
  resolved "https://registry.yarnpkg.com/inert/-/inert-4.2.1.tgz#da743c478a18a8378032f80ada128a28cd2bba93"
  dependencies:
    ammo "2.x.x"
    boom "5.x.x"
    hoek "4.x.x"
    items "2.x.x"
    joi "10.x.x"
    lru-cache "4.1.x"

iron@4.x.x, iron@^4.0.5:
  version "4.0.5"
  resolved "https://registry.yarnpkg.com/iron/-/iron-4.0.5.tgz#4f042cceb8b9738f346b59aa734c83a89bc31428"
  dependencies:
    boom "5.x.x"
    cryptiles "3.x.x"
    hoek "4.x.x"

is-stream@^1.0.1:
  version "1.1.0"
  resolved "https://registry.yarnpkg.com/is-stream/-/is-stream-1.1.0.tgz#12d4a3dd4e68e0b79ceb8dbc84173ae80d91ca44"

isemail@1.x.x:
  version "1.2.0"
  resolved "https://registry.yarnpkg.com/isemail/-/isemail-1.2.0.tgz#be03df8cc3e29de4d2c5df6501263f1fa4595e9a"

isemail@2.x.x:
  version "2.2.1"
  resolved "https://registry.yarnpkg.com/isemail/-/isemail-2.2.1.tgz#0353d3d9a62951080c262c2aa0a42b8ea8e9e2a6"

isomorphic-fetch@^2.1.1:
  version "2.2.1"
  resolved "https://registry.yarnpkg.com/isomorphic-fetch/-/isomorphic-fetch-2.2.1.tgz#611ae1acf14f5e81f729507472819fe9733558a9"
  dependencies:
    node-fetch "^1.0.1"
    whatwg-fetch ">=0.10.0"

items@2.x.x, items@^2.1.1:
  version "2.1.1"
  resolved "https://registry.yarnpkg.com/items/-/items-2.1.1.tgz#8bd16d9c83b19529de5aea321acaada78364a198"

joi@10.x.x, joi@^10.6.0:
  version "10.6.0"
  resolved "https://registry.yarnpkg.com/joi/-/joi-10.6.0.tgz#52587f02d52b8b75cdb0c74f0b164a191a0e1fc2"
  dependencies:
    hoek "4.x.x"
    isemail "2.x.x"
    items "2.x.x"
    topo "2.x.x"

joi@^6.10.1:
  version "6.10.1"
  resolved "https://registry.yarnpkg.com/joi/-/joi-6.10.1.tgz#4d50c318079122000fe5f16af1ff8e1917b77e06"
  dependencies:
    hoek "2.x.x"
    isemail "1.x.x"
    moment "2.x.x"
    topo "1.x.x"

js-tokens@^3.0.0:
  version "3.0.2"
  resolved "https://registry.yarnpkg.com/js-tokens/-/js-tokens-3.0.2.tgz#9866df395102130e38f7f996bceb65443209c25b"

jsonwebtoken@^7.1.9:
  version "7.4.3"
  resolved "https://registry.yarnpkg.com/jsonwebtoken/-/jsonwebtoken-7.4.3.tgz#77f5021de058b605a1783fa1283e99812e645638"
  dependencies:
    joi "^6.10.1"
    jws "^3.1.4"
    lodash.once "^4.0.0"
    ms "^2.0.0"
    xtend "^4.0.1"

jwa@^1.1.4:
  version "1.1.5"
  resolved "https://registry.yarnpkg.com/jwa/-/jwa-1.1.5.tgz#a0552ce0220742cd52e153774a32905c30e756e5"
  dependencies:
    base64url "2.0.0"
    buffer-equal-constant-time "1.0.1"
    ecdsa-sig-formatter "1.0.9"
    safe-buffer "^5.0.1"

jws@^3.1.4:
  version "3.1.4"
  resolved "https://registry.yarnpkg.com/jws/-/jws-3.1.4.tgz#f9e8b9338e8a847277d6444b1464f61880e050a2"
  dependencies:
    base64url "^2.0.0"
    jwa "^1.1.4"
    safe-buffer "^5.0.1"

lodash-es@^4.2.1:
  version "4.17.4"
  resolved "https://registry.yarnpkg.com/lodash-es/-/lodash-es-4.17.4.tgz#dcc1d7552e150a0640073ba9cb31d70f032950e7"

lodash.once@^4.0.0:
  version "4.1.1"
  resolved "https://registry.yarnpkg.com/lodash.once/-/lodash.once-4.1.1.tgz#0dd3971213c7c56df880977d504c88fb471a97ac"

lodash@^4.2.1:
  version "4.17.4"
  resolved "https://registry.yarnpkg.com/lodash/-/lodash-4.17.4.tgz#78203a4d1c328ae1d86dca6460e369b57f4055ae"

loose-envify@^1.0.0, loose-envify@^1.1.0, loose-envify@^1.3.1:
  version "1.3.1"
  resolved "https://registry.yarnpkg.com/loose-envify/-/loose-envify-1.3.1.tgz#d1a8ad33fa9ce0e713d65fdd0ac8b748d478c848"
  dependencies:
    js-tokens "^3.0.0"

lru-cache@4.1.x:
  version "4.1.1"
  resolved "https://registry.yarnpkg.com/lru-cache/-/lru-cache-4.1.1.tgz#622e32e82488b49279114a4f9ecf45e7cd6bba55"
  dependencies:
    pseudomap "^1.0.2"
    yallist "^2.1.2"

meetup-web-mocks@1.0.217:
  version "1.0.217"
  resolved "https://registry.yarnpkg.com/meetup-web-mocks/-/meetup-web-mocks-1.0.217.tgz#a3bd6d458ab8a8b9c38fa96a2125c5b5e74b5330"

mime-db@1.x.x:
  version "1.32.0"
  resolved "https://registry.yarnpkg.com/mime-db/-/mime-db-1.32.0.tgz#485b3848b01a3cda5f968b4882c0771e58e09414"

mimos@^3.0.3:
  version "3.0.3"
  resolved "https://registry.yarnpkg.com/mimos/-/mimos-3.0.3.tgz#b9109072ad378c2b72f6a0101c43ddfb2b36641f"
  dependencies:
    hoek "4.x.x"
    mime-db "1.x.x"

moment@2.x.x:
  version "2.19.4"
  resolved "https://registry.yarnpkg.com/moment/-/moment-2.19.4.tgz#17e5e2c6ead8819c8ecfad83a0acccb312e94682"

ms@^2.0.0:
  version "2.1.1"
  resolved "https://registry.yarnpkg.com/ms/-/ms-2.1.1.tgz#30a5864eb3ebb0a66f2ebe6d727af06a09d86e0a"

nigel@2.x.x:
  version "2.0.2"
  resolved "https://registry.yarnpkg.com/nigel/-/nigel-2.0.2.tgz#93a1866fb0c52d87390aa75e2b161f4b5c75e5b1"
  dependencies:
    hoek "4.x.x"
    vise "2.x.x"

node-fetch@^1.0.1:
  version "1.7.3"
  resolved "https://registry.yarnpkg.com/node-fetch/-/node-fetch-1.7.3.tgz#980f6f72d85211a5347c6b2bc18c5b84c3eb47ef"
  dependencies:
    encoding "^0.1.11"
    is-stream "^1.0.1"

object-assign@^4.1.0, object-assign@^4.1.1:
  version "4.1.1"
  resolved "https://registry.yarnpkg.com/object-assign/-/object-assign-4.1.1.tgz#2109adc7965887cfc05cbbd442cac8bfbb360863"

pez@2.x.x:
  version "2.1.5"
  resolved "https://registry.yarnpkg.com/pez/-/pez-2.1.5.tgz#5ec2cc62500cc3eb4236d4a414cf5a17b5eb5007"
  dependencies:
    b64 "3.x.x"
    boom "5.x.x"
    content "3.x.x"
    hoek "4.x.x"
    nigel "2.x.x"

podium@^1.3.0:
  version "1.3.0"
  resolved "https://registry.yarnpkg.com/podium/-/podium-1.3.0.tgz#3c490f54d16f10f5260cbe98641f1cb733a8851c"
  dependencies:
    hoek "4.x.x"
    items "2.x.x"
    joi "10.x.x"

promise@^7.1.1:
  version "7.3.1"
  resolved "https://registry.yarnpkg.com/promise/-/promise-7.3.1.tgz#064b72602b18f90f29192b8b1bc418ffd1ebd3bf"
  dependencies:
    asap "~2.0.3"

prop-types@^15.5.4:
  version "15.6.0"
  resolved "https://registry.yarnpkg.com/prop-types/-/prop-types-15.6.0.tgz#ceaf083022fc46b4a35f69e13ef75aed0d639856"
  dependencies:
    fbjs "^0.8.16"
    loose-envify "^1.3.1"
    object-assign "^4.1.1"

pseudomap@^1.0.2:
  version "1.0.2"
  resolved "https://registry.yarnpkg.com/pseudomap/-/pseudomap-1.0.2.tgz#f052a28da70e618917ef0a8ac34c1ae5a68286b3"

punycode@^1.4.1:
  version "1.4.1"
  resolved "https://registry.yarnpkg.com/punycode/-/punycode-1.4.1.tgz#c0d5a63b2718800ad8e1eb0fa5269c84dd41845e"

react-helmet@5.2.0:
  version "5.2.0"
  resolved "https://registry.yarnpkg.com/react-helmet/-/react-helmet-5.2.0.tgz#a81811df21313a6d55c5f058c4aeba5d6f3d97a7"
  dependencies:
    deep-equal "^1.0.1"
    object-assign "^4.1.1"
    prop-types "^15.5.4"
    react-side-effect "^1.1.0"

react-side-effect@^1.1.0:
  version "1.1.3"
  resolved "https://registry.yarnpkg.com/react-side-effect/-/react-side-effect-1.1.3.tgz#512c25abe0dec172834c4001ec5c51e04d41bc5c"
  dependencies:
    exenv "^1.2.1"
    shallowequal "^1.0.1"

redux@3.7.2:
  version "3.7.2"
  resolved "https://registry.yarnpkg.com/redux/-/redux-3.7.2.tgz#06b73123215901d25d065be342eb026bc1c8537b"
  dependencies:
    lodash "^4.2.1"
    lodash-es "^4.2.1"
    loose-envify "^1.1.0"
    symbol-observable "^1.0.3"

safe-buffer@^5.0.1:
  version "5.1.1"
  resolved "https://registry.yarnpkg.com/safe-buffer/-/safe-buffer-5.1.1.tgz#893312af69b2123def71f57889001671eeb2c853"

setimmediate@^1.0.5:
  version "1.0.5"
  resolved "https://registry.yarnpkg.com/setimmediate/-/setimmediate-1.0.5.tgz#290cbb232e306942d7d7ea9b83732ab7856f8285"

shallowequal@^1.0.1:
  version "1.0.2"
  resolved "https://registry.yarnpkg.com/shallowequal/-/shallowequal-1.0.2.tgz#1561dbdefb8c01408100319085764da3fcf83f8f"

shot@^3.4.2:
  version "3.4.2"
  resolved "https://registry.yarnpkg.com/shot/-/shot-3.4.2.tgz#1e5c3f6f2b26649adc42f7eb350214a5a0291d67"
  dependencies:
    hoek "4.x.x"
    joi "10.x.x"

statehood@^5.0.3:
  version "5.0.3"
  resolved "https://registry.yarnpkg.com/statehood/-/statehood-5.0.3.tgz#c07a75620db5379b60d2edd47f538002a8ac7dd6"
  dependencies:
    boom "5.x.x"
    cryptiles "3.x.x"
    hoek "4.x.x"
    iron "4.x.x"
    items "2.x.x"
    joi "10.x.x"

subtext@^5.0.0:
  version "5.0.0"
  resolved "https://registry.yarnpkg.com/subtext/-/subtext-5.0.0.tgz#9c3f083018bb1586b167ad8cfd87083f5ccdfe0f"
  dependencies:
    boom "5.x.x"
    content "3.x.x"
    hoek "4.x.x"
    pez "2.x.x"
    wreck "12.x.x"

symbol-observable@^1.0.3:
  version "1.1.0"
  resolved "https://registry.yarnpkg.com/symbol-observable/-/symbol-observable-1.1.0.tgz#5c68fd8d54115d9dfb72a84720549222e8db9b32"

topo@1.x.x:
  version "1.1.0"
  resolved "https://registry.yarnpkg.com/topo/-/topo-1.1.0.tgz#e9d751615d1bb87dc865db182fa1ca0a5ef536d5"
  dependencies:
    hoek "2.x.x"

topo@2.x.x, topo@^2.0.2:
  version "2.0.2"
  resolved "https://registry.yarnpkg.com/topo/-/topo-2.0.2.tgz#cd5615752539057c0dc0491a621c3bc6fbe1d182"
  dependencies:
    hoek "4.x.x"

tough-cookie@2.3.2:
  version "2.3.2"
  resolved "https://registry.yarnpkg.com/tough-cookie/-/tough-cookie-2.3.2.tgz#f081f76e4c85720e6c37a5faced737150d84072a"
  dependencies:
    punycode "^1.4.1"

ua-parser-js@^0.7.9:
  version "0.7.17"
  resolved "https://registry.yarnpkg.com/ua-parser-js/-/ua-parser-js-0.7.17.tgz#e9ec5f9498b9ec910e7ae3ac626a805c4d09ecac"

uuid@^2.0.1:
  version "2.0.3"
  resolved "https://registry.yarnpkg.com/uuid/-/uuid-2.0.3.tgz#67e2e863797215530dff318e5bf9dcebfd47b21a"

vise@2.x.x:
  version "2.0.2"
  resolved "https://registry.yarnpkg.com/vise/-/vise-2.0.2.tgz#6b08e8fb4cb76e3a50cd6dd0ec37338e811a0d39"
  dependencies:
    hoek "4.x.x"

whatwg-fetch@>=0.10.0:
  version "2.0.3"
  resolved "https://registry.yarnpkg.com/whatwg-fetch/-/whatwg-fetch-2.0.3.tgz#9c84ec2dcf68187ff00bc64e1274b442176e1c84"

wreck@12.x.x:
  version "12.5.1"
  resolved "https://registry.yarnpkg.com/wreck/-/wreck-12.5.1.tgz#cd2ffce167449e1f0242ed9cf80552e20fb6902a"
  dependencies:
    boom "5.x.x"
    hoek "4.x.x"

xtend@^4.0.1:
  version "4.0.1"
  resolved "https://registry.yarnpkg.com/xtend/-/xtend-4.0.1.tgz#a5c6d532be656e23db820efb943a1f04998d63af"

yallist@^2.1.2:
  version "2.1.2"
  resolved "https://registry.yarnpkg.com/yallist/-/yallist-2.1.2.tgz#1c11f9218f076089a47dd512f93c6699a6a81d52"
