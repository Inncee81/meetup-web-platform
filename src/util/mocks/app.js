import {
	MOCK_GROUP,
	MOCK_EVENT,
} from './api';

export const MOCK_APP_STATE = {
	app: {
		self: {
			type: 'member',
			value: { id: 1234 }
		},
		group: {
			type: 'group',
			value: MOCK_GROUP
		},
		events: {
			type: 'event',
			value: [
				{ ...MOCK_EVENT, ...({ id: 3456 }) },
				{ ...MOCK_EVENT, ...({ id: 4567 }) }
			]
		}
	},
	auth: {},
	config: {},
	routing: {
		locationBeforeTransitions: {}
	},
};

export const MOCK_ROUTES = {
	path: '/',
	component: {},
	query: () => {}
};

export const MOCK_API_PROBLEM = {
	problem: 'There was an internal problem in the API'
};

export const MOCK_API_RESULT = [{
	ref: 'group',
	type: 'group',
	value: MOCK_APP_STATE.app.group.value
}];

export const MOCK_RENDERPROPS = {
	location: {  // https://github.com/reactjs/history/blob/master/docs/Location.md
		pathname: '/foo',
		search: '',
		state: {},
		action: 'PUSH',
		key: '1234'
	},
	params: {
		urlname: 'foo'
	}
};

export const mockQuery = ({ location, params }) => {
	return {
		type: 'group',
		params: params,
		ref: 'group'
	};
};

export const mockQueryBadType = ({ location, params }) => {
	const type = 'lkajlhsdhljaskliub';
	return { ...mockQuery({ location, params }), type };
};

export const MOCK_AUTH_HEADER = 'Bearer abcdefg';
export const MOCK_DATETIME = new Date().getTime();
export const MOCK_CREDENTIALS = {
	username: 'Michael McGahan',
	pw: 'whatever'
};

export const MOCK_POST_ACTION = {
	type: 'POST_DUMMY',
	payload: {
		query: {},
		onSuccess: result => ({ type: 'MOCK_SUCCESS' }),
		onError: err => ({ type: 'MOCK_ERROR' }),
	}
};
