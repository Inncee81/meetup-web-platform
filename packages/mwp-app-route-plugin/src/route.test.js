import Boom from 'boom';
import { MOCK_RENDER_RESULT } from 'meetup-web-mocks/lib/app';
import { getServer } from 'mwp-test-utils';
import getRoute, { onPreResponse } from './route';

describe('onPreResponse.method', () => {
	it('returns html containing error message', () => {
		const errorMessage = 'foobar';
		const errorCode = 432;
		const request = {
			response: Boom.create(errorCode, errorMessage),
			route: {},
			server: getServer(),
		};
		const replyObj = {
			code() {},
		};
		const spyable = {
			reply: () => replyObj,
		};
		spyOn(replyObj, 'code');
		spyOn(spyable, 'reply').and.callThrough();
		const errorResponse = onPreResponse.method(request, spyable.reply);
		expect(errorResponse).toBe(replyObj);
		const errorMarkup = spyable.reply.calls.mostRecent().args[0];
		expect(errorMarkup).toContain(errorMessage);
		expect(replyObj.code).toHaveBeenCalledWith(errorCode);
	});
	it('returns 500 page', () => {
		const request = {
			response: Boom.create(500, 'foobar'),
			route: {},
			server: getServer(),
		};
		const replyObj = {
			code() {},
		};
		const spyable = {
			reply: () => replyObj,
		};
		spyOn(replyObj, 'code');
		spyOn(spyable, 'reply').and.callThrough();
		const errorResponse = onPreResponse.method(request, spyable.reply);
		expect(errorResponse).toBe(replyObj);
		const errorMarkup = spyable.reply.calls.mostRecent().args[0];
		expect(errorMarkup).toMatchSnapshot();
	});
	it('serves the homepage route', () => {
		const server = getServer();
		const result = 'ok';
		server.route(
			getRoute({
				'en-US': () => Promise.resolve({ statusCode: 200, result }),
			})
		);
		return server
			.inject({ url: '/' })
			.then(response => expect(response.payload).toEqual(result));
	});
});
