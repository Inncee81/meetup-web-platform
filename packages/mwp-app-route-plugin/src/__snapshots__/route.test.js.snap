// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`onPreResponse.method returns 500 page 1`] = `"<!DOCTYPE html><html><body><div style=\\"box-sizing:border-box;font-family:sans-serif;position:fixed;padding:10px;top:0px;left:0px;bottom:0px;right:0px;width:100%;background:rgb(204, 0, 0);color:white;z-index:2147483647;text-align:left;font-size:16px;line-height:1.2;overflow:auto\\" data-reactroot=\\"\\"><div style=\\"font-weight:bold\\">Error<!-- -->: <!-- -->foobar</div><div style=\\"font-family:monospace;margin-top:2em\\"><div style=\\"margin-top:1em\\"><div>Object.&lt;anonymous&gt;</div><div style=\\"font-size:0.8em;color:rgba(255, 255, 255, 0.7)\\"><a href=\\"file:///Users/lori/projects/meetup-web-platform/packages/mwp-app-route-plugin/src/route.test.js\\" style=\\"text-decoration:none;color:rgba(255, 255, 255, 0.7)\\">/Users/lori/projects/meetup-web-platform/packages/mwp-app-route-plugin/src/route.test.js:31:29</a></div></div><div style=\\"margin-top:1em\\"><div>Object.asyncFn</div><div style=\\"font-size:0.8em;color:rgba(255, 255, 255, 0.7)\\"><a href=\\"file:///Users/lori/projects/mup-web/node_modules/jest-jasmine2/build/jasmine-async.js\\" style=\\"text-decoration:none;color:rgba(255, 255, 255, 0.7)\\">/Users/lori/projects/mup-web/node_modules/jest-jasmine2/build/jasmine-async.js:68:30</a></div></div><div style=\\"margin-top:1em\\"><div>resolve</div><div style=\\"font-size:0.8em;color:rgba(255, 255, 255, 0.7)\\"><a href=\\"file:///Users/lori/projects/mup-web/node_modules/jest-jasmine2/build/queueRunner.js\\" style=\\"text-decoration:none;color:rgba(255, 255, 255, 0.7)\\">/Users/lori/projects/mup-web/node_modules/jest-jasmine2/build/queueRunner.js:38:12</a></div></div><div style=\\"margin-top:1em\\"><div>mapper</div><div style=\\"font-size:0.8em;color:rgba(255, 255, 255, 0.7)\\"><a href=\\"file:///Users/lori/projects/mup-web/node_modules/jest-jasmine2/build/queueRunner.js\\" style=\\"text-decoration:none;color:rgba(255, 255, 255, 0.7)\\">/Users/lori/projects/mup-web/node_modules/jest-jasmine2/build/queueRunner.js:31:21</a></div></div><div style=\\"margin-top:1em\\"><div>Promise.resolve.then.el</div><div style=\\"font-size:0.8em;color:rgba(255, 255, 255, 0.7)\\"><a href=\\"file:///Users/lori/projects/mup-web/node_modules/p-map/index.js\\" style=\\"text-decoration:none;color:rgba(255, 255, 255, 0.7)\\">/Users/lori/projects/mup-web/node_modules/p-map/index.js:46:16</a></div></div><div style=\\"margin-top:1em\\"><div>process._tickDomainCallback</div><div style=\\"font-size:0.8em;color:rgba(255, 255, 255, 0.7)\\"><a href=\\"file://internal/process/next_tick.js\\" style=\\"text-decoration:none;color:rgba(255, 255, 255, 0.7)\\">internal/process/next_tick.js:228:7</a></div></div></div></div></body></html>"`;
